### DISCUSSION TITLE
uniapp中引入jsmind时, 思维导图的线 和 结点 会分开显示, 不在一起

### DISCUSSION THREAD

**Discussion by flightsnow** [hizzgdev/jsmind#536] [General]
使用uniapp引入了jsmind, 想实现jsmind.online中显示的app效果. 
但是最终效果是: 思维导图的线单独显示在上面, 所有结点单独显示在下面. 
样式代码为: 
```
<style scoped>
	.scrollable-container {
	  width: 100%;
	  overflow-x: auto;
	}
    .jsmind-container {
        height: 700px;
        border: solid 1px #ccc;
    }
</style>

<template>
	<div class="scrollable-container">
		<div ref="jsmind_container" class="jsmind-container"></div>
	</div>
</template>
```

![搜狗截图20231118021902](https://github.com/hizzgdev/jsmind/assets/12780080/7abb8559-fa78-4fe0-b23e-909b8ed583f2)

**Comment by flightsnow** [comment]
template中代码为:

`<template>
	<div class="scrollable-container">
		<div ref="jsmind_container" class="jsmind-container"></div>
	</div>
</template>`

(github显示<template>代码样式好像有问题, 已经加了代码块, 但是会把换行吃掉... )

**Comment by hizzgdev** [comment]
我不是很确定问题在哪里。不过你要不要看一下最终它 render 出来的 html 结构是什么样的。以及跟 jsmind.online 里的样式对比一下。主要是 `jsmind_container` 的样式，看看是不是有啥不一样的地方。
我直观的感觉应该是没有设置 `width` 造成的，但也不知道在这个页面里有多少样式对 `jsmind_container` 有影响。

**Comment by flightsnow** [comment]
好的, 感谢您的建议. 之前是有加width的, 但是加了width且, 思维导图的线也是分开的, 只不过一个在左边, 一个在右边. 后面我就将宽度去掉试了下, 然后它就变成一个在上面一个在下面了.  这个页面中只有上面列出来两个样式, 且尝试了只保留jsmind_container一个样式的情况下 , 线和结点也是分开展示的.
另外我想请教一下, jsmind.online上面的样式可以给出来参考一下嘛? more samples中没有给出app下的sample. 我通过jsmind.online上面的检查样式查看, 看的不太明白.  
所以我是基于more sample中的给出的样例进行修改的. 而sample中给出的样例, jsmind_container所用的div都是给了固定宽度, (宽高改用wrapcontent也没有效果)  无法做到jsmind.online中的那种宽高会根据导图变化而自适应的效果~

**Comment by hizzgdev** [comment]
jsmind.online 上的代码你可以在浏览器里直接看到，在检查样式里看不出来，也可以把样式表下载到电脑上再看。
jsmind.online 上宽高自适应是监听了 window 的 resize 事件后，重新设置 width/height 实现的。

uniapp 这个如果你这个调试好的话，非常欢迎分享到 jsmind-samples 这个项目里来。现在前端各种框架确实太多了。

**Comment by maskj4** [comment]
我也是用uniapp调试的canvas缺少position: absolute加上就行了
